src/main/developerTools/logHandler.js

```diff
@@ -1,82 +0,0 @@
-const { createWriteStream } = require('fs')
-const mkdirp = require('mkdirp')
-const { resolve: pathResolve } = require('path')
-const { app } = require('electron')
-const windows = require('../windows')
-const { getLogsPath } = require('../../application-constants')
-var events = require('events')
-var eventEmitter = new events.EventEmitter()
-var logDB = []
-var wstream
-var wsUpState = false
-var fullLogFilePath
-const CONFIG_LOG_MAIN_TO_CHROME_CONSOLE = false
-/**
- * Internal logger - please don't call directly unless you know what your doing
- * @param {string} timestamp timestamp
- * @param {string} channel The part/module where the message was logged. Like 'Tanslations'
- * @param {Number} level The level of importance. 0 for debug, 1 for info, 2 for warning, 3 for error, 4 for critical error
- * @param {string} message the message (human readable)
- * @param {string} errorCode (optional) machine readable error code (string in snake_case)
- * @param {any|string} payload (optional) JSON payload or any other object that can be converted to a JSON string
- * @param {StackFrame} stacktrace (optional) for errors -> gets converted to json
- */
-function log (timestamp, channel, level, message, errorCode, payload, stacktrace) {
-  logDB.push({ channel, level, message, errorCode, payload, stacktrace, timestamp })
-
-  if (payload) {
-    var payloadForLogfile = JSON.stringify(payload)
-    if (payloadForLogfile.charAt(0) !== '"') {
-      payloadForLogfile = JSON.stringify(payloadForLogfile)
-    }
-  }
-
-  const logString = `${timestamp}|=|=|"${channel}"|=|=|${level}|=|=|"${message}"|=|=|"${errorCode || ''}"|=|=|${payloadForLogfile || ''}|=|=|${JSON.stringify(stacktrace || '')}`
-    .replace(/\t/g, '  ').replace(/\n|(:?\r\n)/g, '') // Cleanup
```

Comment:
> Maybe we should keep this "clean up of the json" to be on the safe side. When we keep it it should be in the map function that converts the args to json.

This is being overly anal imo. For example:

```js
var s = 'wooooooo\tfoobar'
console.log(JSON.stringify(s))
// "wooooooo\tfoobar"
```

Since `JSON.stringify` escapes the `\t`. Same goes for `\n`.

Reply:
Your right, JSON.stringify should work properly.I woried about the strange results that one get when the string gets stringified multiple times - But thats too unlikely to happen.